micrometer-core/src/test/java/io/micrometer/core/instrument/simple/SimpleMeterRegistryTest.java

```diff
@@ -101,6 +109,133 @@ public void newFunctionCounterWhenCountingModeIsStepShouldReturnStepFunctionCoun
+
+        Meter.builder("default.meter", Type.GAUGE,
+                Arrays.asList(
+                        new Measurement(() -> 7.7, Statistic.VALUE),
+                        new Measurement(() -> 8.8, Statistic.VALUE)
+                ))
+                .baseUnit("beans")
+                .register(registry);
+
+        final LongTaskTimer.Sample loopSample = taskTimer.start();
+
+        for (int i = 0; i < 300; i++) {
+            responseTimer.record(Duration.ofMillis(i));
+            requestSizeCounter.increment(i);
+            responseSizeSummary.record(i);
+        }
+
+        loopSample.stop();
+
+        testRunTime.set(200);
+        assertThat(registry.toString()).isEqualTo("\n"
+                + "FunctionCounter cache.evictions count=10.000000\n"
+                + "FunctionTimer cache.latency\n"
+                + "        count = 10.000000\n"
+                + "    totalTime = 0.300000\n"
+                + "         mean = 0.030000\n"
+                + "Meter default.meter\n"
+                + "        VALUE = 7.700000\n"
+                + "        VALUE = 8.800000\n"
+                + "Counter request.size count=44850.000000\n"
+                + "DistributionSummary response.size\n"
+                + "        count = 300\n"
+                + "        total = 44850.000000\n"
+                + "         mean = 149.500000\n"
+                + "          max = 299.000000\n"
+                + "    percentileValues:\n"
+                + "        75.0 <= 231.9375\n"
+                + "        90.0 <= 271.9375\n"
+                + "        95.0 <= 287.9375\n"
+                + "        99.0 <= 303.9375\n"
+                + "       99.99 <= 303.9375\n"
+                + "    histogramCounts:\n"
+                + "       250.0 <= 256.0\n"
+                + "       500.0 <= 300.0\n"
+                + "      1000.0 <= 300.0\n"
+                + "      2000.0 <= 300.0\n"
+                + "      4000.0 <= 300.0\n"
+                + "      6000.0 <= 300.0\n"
+                + "Gauge response.size.histogram [tag(le=250)] value=256.000000\n"
+                + "Gauge response.size.histogram [tag(le=500)] value=300.000000\n"
+                + "Gauge response.size.histogram [tag(le=1000)] value=300.000000\n"
+                + "Gauge response.size.histogram [tag(le=2000)] value=300.000000\n"
+                + "Gauge response.size.histogram [tag(le=4000)] value=300.000000\n"
+                + "Gauge response.size.histogram [tag(le=6000)] value=300.000000\n"
+                + "Gauge response.size.percentile [tag(phi=0.9)] value=271.937500\n"
+                + "Gauge response.size.percentile [tag(phi=0.75)] value=231.937500\n"
+                + "Gauge response.size.percentile [tag(phi=0.95)] value=287.937500\n"
+                + "Gauge response.size.percentile [tag(phi=0.99)] value=303.937500\n"
+                + "Gauge response.size.percentile [tag(phi=0.9999)] value=303.937500\n"
```

Comment:
Let's simplify the tag from `[tag(phi=123)]` to something simpler. Perhaps something like `{phi=123;action=something}`

Reply:
Do you want a space after `;`?

`{phi=123; action=something}`

Reply:
Yeah a space would help reading